src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java

```diff
@@ -248,10 +249,23 @@ public void run() {
                         } catch (ClosedChannelException e) {
                             onRecvClosed();
                             return;
-                        } catch (EOFException e) {
+                        } catch (IOException e) {
```

Comment:
I think we may have a timeout set [here](https://github.com/jenkinsci/remoting/blob/a4f5af3617277b53c0b350d64423b7b7009bfc8c/src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpoint.java#L183) - so this would introduce a regression IIUC.